Remove Participation module

stbnrivas opened the pull request:
Closes #3826 

## :v: What does this PR do?

remove module participations and its dependencies into other modules

- gobierto cms
- gobierto_calendars | because process and event (pg_search)

stbnrivas committed:
WIP removing files [skip ci]

stbnrivas commented:
Hi!, @ferblape about 

https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/models/gobierto_cms/page_test.rb#L62-L70

about this test this should be (2 preferred)
1. removed 
2. replaced for another model ¿why model can I use?

ferblape commented:
You can remove that test and the method destroyable? in cms page model

On Thu, 1 Jul 2021 at 10:00, stbnrivas ***@***.***> wrote:

> Hi!, @ferblape <https://github.com/ferblape> about
>
>
> https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/models/gobierto_cms/page_test.rb#L62-L70
>
> about this test this should be (2 preferred)
>
>    1. removed
>    2. replaced for another model ¿why model can I use?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PopulateTools/gobierto/pull/3909#issuecomment-872017780>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAEJUBNVHQMIQI7QBWNTTLTVQOCPANCNFSM47T5TJ7Q>
> .
>


-- 
Fernando Blat
***@***.***
+34 660825001

Populate / Tools for civic engagement
https://populate.tools

Project stories
twitter.com/populate_ & populate.tools/blog

stbnrivas commented:
what's about `GobiertoAttachment::AttachmentDocumentsController`?  Does it make sense without GobiertoParticipations::Process? 

https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/app/controllers/gobierto_attachments/attachment_documents_controller.rb#L32-L41

thx

ferblape commented:
Yes, they are used in the cms page model.

Remove the assignment to the current process in that method and leave the
rest

On Thu, 1 Jul 2021 at 10:33, stbnrivas ***@***.***> wrote:

> what's about attachment? Does it make sense without
> GobiertoParticipations::Process?
>
>
> https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/app/controllers/gobierto_attachments/attachment_documents_controller.rb#L32-L41
>
> thx
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PopulateTools/gobierto/pull/3909#issuecomment-872044195>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAEJUCWT4AG345BTKWGGPLTVQR7HANCNFSM47T5TJ7Q>
> .
>


-- 
Fernando Blat
***@***.***
+34 660825001

Populate / Tools for civic engagement
https://populate.tools

Project stories
twitter.com/populate_ & populate.tools/blog

stbnrivas commented:
@blat I have a problem with a test that before works and not now ... I am missing something...

https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/models/gobierto_common/term_test.rb#L10-L24
but the problem is that this fixtures(culture_term, center_term, marvel_term)  don't have dependent into sublevel


https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/models/gobierto_common/term_test.rb#L71-L75


an example of dependant level with animals and dog fixtures level is (note line 19 below)
https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/fixtures/gobierto_common/terms.yml#L3-L20


but into all term.yml there are single culture_term
https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/fixtures/gobierto_common/terms.yml#L63


1. I propose change:
```ruby
# change this
  def test_destroy_of_term_with_dependencies
    terms_with_dependencies.each do |_, term|
      refute term.destroy
    end
  end

# to this
  def test_destroy_of_term_with_dependencies
    refute root_level_term.destroy
  end
```


But I have a worse problem the decorator don't detect terms collection here

https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/app/models/gobierto_common/term.rb#L45-L53


TermDependendentResourceDecorator is not enough. I need add some changes at:
https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/app/models/gobierto_common/term.rb#L45-L53

2. I propose add new case to dependant
```
def destroy
      dependent_resources_decorator = TermDependentResourcesDecorator.new(self)
      if dependent_resources_decorator.has_dependent_resources?
        errors.add(:base, :has_dependent_resources_html, dependencies_list: dependent_resources_decorator.dependencies_list)
        return false
      elsif !terms.empty?
        errors.add(:base, :has_dependent_resources_html, dependencies_list: "term_id")
        return false
      end

      super
    end
```

What am I missing?

stbnrivas commented:
should I delete (full) or refactor this decorator and its test? . I am not sure about because it looks like a feature into GobiertoCMS designed to be used by GobiertoParticipations. 

So with participation deleted. Does it keeping sense?

https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/decorators/gobierto_cms/page_decorator_test.rb#L27-L33

thx

ferblape commented:
Hmmmm, not exactly, that decorator is used to customize GobiertoCms::Pages, which are used in participation module but also in other modules. So leave the decorator and update the test to remove the relative to themes.

stbnrivas commented:
what about this test? should I delete it or where should I redirect path instead of  `admin_participation_process_pages_path(process.id)`

https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/integration/gobierto_admin/gobierto_common/update_cms_pages_collection_test.rb#L5-L47

thx

ferblape commented:
Delete it, please

On Fri, 2 Jul 2021 at 16:49, stbnrivas ***@***.***> wrote:

> what about this test? should I delete it or where should I redirect path
> instead of admin_participation_process_pages_path(process.id)
>
>
> https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/integration/gobierto_admin/gobierto_common/update_cms_pages_collection_test.rb#L5-L47
>
> thx
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PopulateTools/gobierto/pull/3909#issuecomment-873056019>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAEJUBN4URFWDEJYR3ANQDTVXGZHANCNFSM47T5TJ7Q>
> .
>


-- 
Fernando Blat
***@***.***
+34 660825001

Populate / Tools for civic engagement
https://populate.tools

Project stories
twitter.com/populate_ & populate.tools/blog

stbnrivas commented:
hi, just to confirm, the test into this folder [https://github.com/PopulateTools/gobierto/tree/master/test/integration/gobierto_admin/gobierto_attachments](https://github.com/PopulateTools/gobierto/tree/master/test/integration/gobierto_admin/gobierto_attachments) are dependent of some fixtures of GobiertoParticipations.  

Should I refactor or delete?

thx

ferblape commented:
refactor, gobierto attachments are out of the affected modules

On Mon, 5 Jul 2021 at 16:42, stbnrivas ***@***.***> wrote:

> hi, just to confirm the test into this folder
> https://github.com/PopulateTools/gobierto/tree/master/test/integration/gobierto_admin/gobierto_attachments
> are dependent of some fixtures of GobiertoParticipations.
>
> Should I refactor or delete?
>
> thx
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PopulateTools/gobierto/pull/3909#issuecomment-874165041>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAEJUHRHC63IXZ4BFYKPJLTWHAG5ANCNFSM47T5TJ7Q>
> .
>


-- 
Fernando Blat
***@***.***
+34 660825001

Populate / Tools for civic engagement
https://populate.tools

Project stories
twitter.com/populate_ & populate.tools/blog

stbnrivas commented:
https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/db/migrate/20170719091606_add_type_to_processes.rb#L3-L7

After the module deletion the class `GobiertoParticipation::Process` is undefined and migration will fail. How I should fix this?
- keep migration implicated but surrounded by `if defined? `
- removing the content of migration method affected?

Just to say,  there are 4 cases of migrations like this 

Thx

ferblape commented:
Good point, using `if defined?` sounds good to me

stbnrivas commented:
Should join fixtures and fixtures_archive? at least which that are still in use ... or I'm missing something?

https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/fixtures/gobierto_cms/pages.yml#L1-L4

https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/fixtures/gobierto_cms/section_items.yml#L1

thx!

stbnrivas commented:
So attachment and participation have a set of test cases which I need remove or rewrite as new feature. Some details:


https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/integration/gobierto_admin/gobierto_attachments/file_attachment_index_test.rb#L40

https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/integration/gobierto_admin/gobierto_common/collections/edit_collection_test.rb#L46

https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/integration/gobierto_admin/gobierto_common/collections/show_collection_test.rb#L35

https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/integration/gobierto_admin/gobierto_common/collections/show_collection_test.rb#L46

before deletion module attachment show the collection and attachments that belongs to `GobiertoParticipation::Process`. These process can be archived with paranoid and then these collections were not show.

I think that this behavior don't apply to attachment collections which don't belongs to Process:
-  I searched examples into staging `admin/attachments/file_attachments`
-  then archive all attachment into a collection 
-  the collection it keep showing (I think that it's right)


therefore this test should be removed, or I am missing something? 
thx

ferblape commented:
What I'd do is to update the collections of the tests to different ones. In
test/fixtures/gobierto_common/collections.yml you have more available, you
could use site_pages, site_news (if you want to use the site Madrid)

On Wed, 7 Jul 2021 at 12:42, stbnrivas ***@***.***> wrote:

> So attachment and participation have a set of test cases which I need
> remove or rewrite as new feature. Some details:
>
>
> https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/integration/gobierto_admin/gobierto_attachments/file_attachment_index_test.rb#L40
>
>
> https://github.com/PopulateTools/gobierto/blob/91e9cfee11ab6bb55990e9ed38dae1a5612e6391/test/integration/gobierto_admin/gobierto_common/collections/edit_collection_test.rb#L46
>
> before deletion module attachment show the collection and attachments that
> belongs to GobiertoParticipation::Process. These process can be archived
> with paranoid and then these collections were not show.
>
> I think that this behavior don't apply to attachment collections which
> don't belongs to Process:
>
>    - I searched examples into staging admin/attachments/file_attachments
>    - then archive all attachment into a collection
>    - the collection it keep showing (I think that it's right)
>
> therefore this test should be removed, or I am missing something?
> thx
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PopulateTools/gobierto/pull/3909#issuecomment-875497964>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAEJUHVKEI5WPQVUOFBJBLTWQVSRANCNFSM47T5TJ7Q>
> .
>


-- 
Fernando Blat
***@***.***
+34 660825001

Populate / Tools for civic engagement
https://populate.tools

Project stories
twitter.com/populate_ & populate.tools/blog

stbnrivas commented:
that was my first step. OkI will separate into different commit to speak about later 

thx

stbnrivas committed:
update file & passing test

stbnrivas committed:
update files

stbnrivas committed:
fixing test

stbnrivas committed:
fixing test

stbnrivas committed:
fixing test

stbnrivas committed:
fixing test

stbnrivas committed:
restore fixtures archive

stbnrivas commented:
I'm having problems passing test of  gobierto_engine_gencat so I found this js errors, into my local env and also into staging

![Screenshot from 2021-07-23 08-59-48](https://user-images.githubusercontent.com/288355/126747752-ad25a0f1-b46f-4623-bbfe-3a5eb4a74a3a.png)

stbnrivas committed:
fix forgotten conflict after rebase

ferblape commented:
@entantoencuanto please continue the work of Esteban in this PR, he's commenting about some errors in gencat engine tests, but it's weird because participation and gencat are not related and don't share code.

entantoencuanto committed:
Fix test

entantoencuanto committed:
Recover current module name in css classes

The class with the module name is used by some engines to select
elements of the DOM

entantoencuanto committed:
Remove unused methods

entantoencuanto committed:
Refactor tests

entantoencuanto committed:
Remove participation elements from seeds and fixtures

entantoencuanto committed:
Fix memoized methods

entantoencuanto committed:
Remove remaining references to participation
